### PR TITLE
[cms-v2] Add subcontractor invoices

### LINKED ISSUE #1692: [cms-v2] Invoice submission for SubContractors
The PR #1566 adds the subcontractor invoice submission for cms V1, and it should also be added to V2.

V1 Issue Reference: #1471

### PR DESCRIPTION
**Pull request by thi4go**
This PR also sneakes in a fix for missing policy argument on reset invoice form data function call.

closes #1692

### CODE REVIEW

File: src/componentsv2/InvoiceForm/validation.js [hunk 1]
```diff
@@ -64,6 +64,16 @@ export const invoiceValidationSchema = ({
               ...yupFieldMatcher("Description", invoicefieldsupportedchars)
             ),
           labor: Yup.number().min(0),
+          subuserid: Yup.string().when("type", (type, schema) =>
+            +type === 4 ? schema.required("required") : schema
+          ),
+          subrate: Yup.number().when("type", (type, schema) =>
+            +type === 4
+              ? schema
+                  .min(500, "must be greater or equal 5")
```
Comment (amassarwi) [inline src/componentsv2/InvoiceForm/validation.js]: Should update the error message

File: src/componentsv2/InvoiceForm/validation.js [hunk 2]
```diff
@@ -64,6 +64,16 @@ export const invoiceValidationSchema = ({
...
+                  .max(50000, "must be less or equal 500")
```
Comment (amassarwi) [inline src/componentsv2/InvoiceForm/validation.js]: Same here

File: src/componentsv2/InvoiceForm/validation.js [hunk 3]
```diff
@@ -64,6 +64,16 @@ export const invoiceValidationSchema = ({
               ...yupFieldMatcher("Description", invoicefieldsupportedchars)
             ),
           labor: Yup.number().min(0),
+          subuserid: Yup.string().when("type", (type, schema) =>
+            +type === 4 ? schema.required("required") : schema
```
Comment (tiagoalvesdulce) [inline src/componentsv2/InvoiceForm/validation.js]: why are you coercing type here and not coercing it in the `processSubtotalColChange` file?

Comment (thi4go) [reply src/componentsv2/InvoiceForm/validation.js]: Fixed this issue mate. Turns out we were validating the type field as a string, which is not correct. Nice catch

**Pull request merged**